### PR TITLE
feat(vmix): add support for automatically changing DVE box inputs

### PR DESCRIPTION
**Pull request by alvancamp**
This PR allows the blueprints to automatically set the DVE box inputs for vMix setups. By default, it expects the DVE to be Input 6, but this can be changed in the Layer Mappings and the README has been updated to explain this process.

### CODE REVIEW

**Review by baltedewit — CHANGES_REQUESTED** [review]
Looks good in principle, I think there's two areas of improvement

- Inputs are defined in a config table so it's a bit weird if the user has to manually edit the mappings only for the VMix DVE. Using multiple inputs would also give the possibility of having multiple layouts as those can't be changed over the API, but that's something for later probably.
- The DVE part adapter always adds timeline objects for both VMix and Atem now, it would be better if it selected which objects to add depending on the studio config.

File: src/showstyle0/part-adapters/dve.ts [hunk 1]
```diff
@@ -60,6 +67,67 @@ export function generateDVEPart(context: PartContext, part: PartProps<DVEProps>)
...
+						2: boxes[1]?.source ?? -1,
+					},
+				},
+			})
+		)
+	} else {
+		assertUnreachable(config.visionMixerType)
+	}
+
+	context.logDebug('timeline objects: ' + JSON.stringify(dvePieceTimelineObjects, undefined, 2))
```
Comment (baltedewit) [inline src/showstyle0/part-adapters/dve.ts:129]: Stray log line

File: src/showstyle0/part-adapters/dve.ts [hunk 2]
```diff
@@ -72,43 +140,69 @@ export function generateDVEPart(context: PartContext, part: PartProps<DVEProps>)
...
+		retainPieceTimelineObjects.push(
+			literal<TSR.TimelineObjAtemSsrcProps>({
+				id: '',
+				enable: { while: 1 },
+				priority: 0.5,
+				layer: AtemLayers.AtemSuperSourceProps,
+				content: {
+					deviceType: TSR.DeviceType.ATEM,
+					type: TSR.TimelineContentTypeAtem.SSRCPROPS,
+					ssrcProps: {
```
Comment (baltedewit) [inline src/showstyle0/part-adapters/dve.ts:158]: I think these should be the same props as the normal piece.

**Review by baltedewit — CHANGES_REQUESTED** [review]
Some small things, looks good in general!

File: src/showstyle0/part-adapters/dve.ts (hunk 1 above)
Comment (alvancamp) [reply src/showstyle0/part-adapters/dve.ts]: Removed

File: src/showstyle0/part-adapters/dve.ts (hunk 2 above)
Comment (alvancamp) [reply src/showstyle0/part-adapters/dve.ts]: Fixed

File: src/studio0/migrations/mappings-defaults.ts [hunk 3]
```diff
@@ -150,35 +150,57 @@ export const AtemMappings = literal<BlueprintMappings>({
 	}),
 })
 
-export const VMixMappings = literal<BlueprintMappings>({
-	[VMixLayers.VMixMeProgram]: literal<TSR.MappingVMixAny & BlueprintMapping>({
-		device: TSR.DeviceType.VMIX,
-		deviceId: 'vmix0',
-		lookahead: LookaheadMode.NONE,
+export function getDyanmicVMixMappings(vmixSources: StudioConfig['vmixSources']): BlueprintMappings {
```
[the remark this answers is not in the document]
Comment (alvancamp) [reply src/studio0/migrations/mappings-defaults.ts]: Fixed

**Pull request merged**